culinary-canvas/capstone-project/backend/index.js

```diff
@@ -57,6 +88,187 @@ const storage = new CloudinaryStorage({
+            Object.keys(healthInfo.nutrients).forEach(nutrient => {
+                if (!totals[nutrient]){
+                    totals[nutrient] = 0;
+                }
+                totals[nutrient]  += healthInfo.nutrients[nutrient]
+            });
+            return totals
+        }, {});
+
+
+        const lowestNutrient = findLowestNutrient(nutrientTotals);
+        const recipesHighInNutrient = await fetchRecipesHighInNutrient(lowestNutrient)
+
+        return{
+            subject: 'Try this recipe!',
+            message: `We noticed you needed more ${lowestNutrient}. So we're suggesting these recipes: `,
+            recipes: recipesHighInNutrient,
+            nutrientTotals : nutrientTotals
+        };
+    }catch(error){
+        console.error('Error fetching user notifications: ', error);
+        return{
+            subject: 'Error',
+            message: 'There was an error fetchig your notifications.',
+            recipes: [],
+            nutrientTotals: {}
+        }
+    }
+
+};
+
+//send notifications to a user
+const sendNotification = async(userId) => {
+    try{
+        const notification = await fetchUserNotifications(userId)
+
+        const recipesHighInNutrient = await notification.recipes
+        await prisma.notification.create({
+            data:{
+                userId: userId,
+                subject: notification.subject,
+                message: notification.message,
+                recipes: recipesHighInNutrient,
+                read: false,
+            }
+        })
+        io.to(userId.toString()).emit('notification', notification);
+    }
+    catch(error){
+        console.error('Error sending notifications', error)
+    }
+};
+
+app.get('/notifications/:userId', async(req, res) => {
+const {userId} = req.params;
+    try{
+        const notifications = await prisma.notification.findMany({
+            where: {userId : parseInt(userId)},
+            orderBy: { createdAt: 'desc'}
+        });
+        res.status(200).json(notifications)
+    }catch(error){
+        console.error("Error fetching notifications: ", error)
+        res.status(500).json({error : 'Failed to fetch notifications.'})
+    }
+})
+
+app.post('/notifications/read', async(req, res) => {
+    const {notificationId} = req.body;
+    try{
+        await prisma.notification.update({
+            where: {id: notificationId},
+            data: {read: true}
+        });
+        res.status(200).json({message: 'Notification marked as read'})
+    }catch(error){
+        console.error('Error marking notification as read: ', error);
+        res.status(500).json({error: 'Failed to mark notifications as read.'})
+    }
+})
+
+
+
+//to send the notifcation every 12am
+cron.schedule('0 0 * * *', async() => {
+
+    try{
+        const users = await prisma.users.findMany();
+        for (const user of users){
+            await sendNotification(user.id)
+        }
```

Comment:
is there a way to map through the users asynchronously? Generally 'await' in a loop is bad practice.